integration-tests/deploy_xgboost_models.py

```diff
@@ -0,0 +1,134 @@
+from __future__ import absolute_import, print_function
+import os
+import sys
+import requests
+import json
+import numpy as np
+import time
+import logging
+import xgboost as xgb
+import pickle
+
+cur_dir = os.path.dirname(os.path.abspath(__file__))
+
+sys.path.insert(0, os.path.abspath('%s/util_direct_import/' % cur_dir))
+from util_package import mock_module_in_package as mmip
+
+from test_utils import (create_docker_connection, BenchmarkException, headers,
+                        log_clipper_state)
+cur_dir = os.path.dirname(os.path.abspath(__file__))
+sys.path.insert(0, os.path.abspath("%s/../clipper_admin" % cur_dir))
+from clipper_admin.deployers.deployer_utils import save_python_function
+
+logging.basicConfig(
+    format='%(asctime)s %(levelname)-8s [%(filename)s:%(lineno)d] %(message)s',
+    datefmt='%y-%m-%d:%H:%M:%S',
+    level=logging.INFO)
+
+logger = logging.getLogger(__name__)
+
+app_name = "xgboost-test"
+model_name = "xgboost-model"
+
+def deploy_and_test_model(clipper_conn,
+                          model,
+                          version,
+                          predict_fn,
+                          link_model=False):
+    serialization_dir = save_python_function(model_name, predict_fn)
+    
+    base_image = 'clipper/python-closure-container:develop'
+    clipper_conn.build_and_deploy_model(model_name, version, "integers",
+                        serialization_dir, base_image, pkgs_to_install=['xgboost'])
```

Comment:
You shouldn't save the function and build the model directly. Use the Python closure deployer to do this for you. That's exactly what it is meant for.